### PR TITLE
Fix Lossless+Progressive JXL and add debug options.

### PR DESCRIPTION
**Pull request by veluca93**
Setting CODEC_DEBUG in the JXL Makefile will produce a debug build (with
error messages enabled).

### CODE REVIEW

File: codecs/jxl/enc/jxl_enc.cpp [hunk 1]
```diff
@@ -57,10 +57,13 @@ val encode(std::string image, int width, int height, JXLOptions options) {
...
   }
 
-  if (options.progressive) {
+  if (options.progressive && !cparams.modular_mode) {
     cparams.qprogressive_mode = true;
     cparams.progressive_dc = 1;
     cparams.responsive = 1;
+  } else {
+    cparams.responsive = 1;
+    cparams.progressive_mode = true;
```
Comment (jakearchibald) [inline codecs/jxl/enc/jxl_enc.cpp]: Isn't this setting progressive mode even if `options.progressive` is false?

Comment (veluca93) [reply codecs/jxl/enc/jxl_enc.cpp]: ... Yes, thanks for catching that...

**Comment by jakearchibald** [comment]
@RReverser can you take a look at the makefile stuff here?

**Comment by jakearchibald** [comment]
This should be merged into `dev`, but I'm happy to take care of that.

**Comment by veluca93** [comment]
Rebased in #848

**Pull request closed without merging**